scripts/workspace_deploy.py

```diff
@@ -81,13 +81,20 @@
     version = getattr(config, 'conversation_version')
     workspacesUrl += '?version=' + version
 
+    # Make sure intents, entities and dialog_nodes fields are present even when empty to make sure the old items get erased.
+    if 'intents' not in workspace: workspace['intents'] = []
+    if 'entities' not in workspace: workspace['entities'] = []
+    if 'dialog_nodes' not in workspace: workspace['dialog_nodes'] = []
+
     # create/update workspace
     response = requests.post(workspacesUrl, auth=(username, password), headers={'Content-Type': 'application/json'}, data=json.dumps(workspace, indent=4))
     responseJson = response.json()
 
     # check errors during upload
     if 'error' in responseJson:
         eprintf('Cannot upload conversation workspace\nERROR: %s\n', responseJson['error'])
+        if 'errors' in responseJson and len(responseJson['errors']) > 0:
+            eprintf('\t explanation: %s\n', responseJson['errors'][0])
```

Comment:
pass more specific info about an error. "duplicate value in entity XXX" is better then "upload failed" even in non-verbose mode.

Reply:
Done in more generic way.